indexer/src/main/java/org/uniprot/store/indexer/uniprotkb/converter/UniProtEntryCrossReferenceConverter.java

```diff
@@ -105,8 +81,31 @@ private void convertXref(List<UniProtKBCrossReference> references, UniProtDocume
             }
         }
         document.d3structure = d3structure;
-        if (d3structure && !document.proteinsWith.contains("3dstructure")) {
-            document.proteinsWith.add("3dstructure");
+        if (d3structure && !document.proteinsWith.contains(ProteinsWith.D3_STRUCTURE.getValue())) {
+            document.proteinsWith.add(ProteinsWith.D3_STRUCTURE.getValue());
+        }
+    }
+
+    private void convertProteomes(UniProtDocument document, UniProtKBCrossReference xref) {
+        document.proteomes.add(xref.getId());
+        if (xref.hasProperties()) {
+            document.proteomeComponents.addAll(
+                    xref.getProperties().stream()
+                            .map(Property::getValue)
+                            .collect(Collectors.toSet()));
+        }
+    }
+
+    private void convertEmbl(
+            UniProtDocument document, UniProtKBCrossReference xref, String dbname) {
+        if (xref.hasProperties()) {
+            Optional<String> proteinId =
+                    xref.getProperties().stream()
+                            .filter(property -> property.getKey().equalsIgnoreCase("ProteinId"))
```

Comment:
is "ProteinId" a constant in the model that we can reference? I'm thinking, if someone changes the model, then this filter will fail silently and give unexpected results